### PR TITLE
Add exclude repo functionality for azure devops and gerrit

### LINKED ISSUE #60346: Support for Azure DevOps repositories missing for "Exclude Repository" button
- **Sourcegraph version:** 5.2
- **Reported by:** https://github.com/sourcegraph/accounts/issues/4142

#### Steps to reproduce:

1. Connect to an Azure DevOps code host
2. Sync at least one repository
3. Go to the repository settings page for a repository from Azure DevOps
4. In the `Options` tab, click on `Exclude repository`

#### Expected behavior:

The repository is excluded

#### Actual behavior:

image

See https://sourcegraph.com/github.com/sourcegraph/sourcegraph@2a8670bc830640f15022d608a1e0cb08a0b7a791/-/blob/cmd/frontend/backend/external_services.go?L335:20-335:38 => https://sourcegraph.com/github.com/sourcegraph/sourcegraph@2a8670bc830640f15022d608a1e0cb08a0b7a791/-/blob/cmd/frontend/backend/external_services.go?L382:6-382:24 (missing azure devops)

If you would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are no [SLAs](https://sourcegraph.com/support/) on issues like there are for support requests).

### PR DESCRIPTION
**Pull request by pjlast**
Closes #60346 

Adds the missing cases for Azure DevOps and Gerrit so that repos can be excluded through the UI

## Test plan

Add unit test and verify manually

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Nice! Were you able to verify this behavior with an actual devops connection?

Side-q while we're here: Are there any other code hosts missing here that otherwise support exclusion?

Also: Maybe a changelog entry?

**Comment by pjlast** [comment]
@eseliger  Yep I tested this locally with an actual DevOps connection :) 

I'll have a look at other code hosts

File: cmd/frontend/backend/external_services_test.go [hunk 1]
```diff
@@ -44,6 +46,13 @@ func TestAddRepoToExclude(t *testing.T) {
 			initialConfig:  `{"accessKeyID":"accessKeyID","gitCredentials":{"password":"","username":""},"region":"","secretAccessKey":""}`,
 			expectedConfig: `{"accessKeyID":"accessKeyID","exclude":[{"name":"test"}],"gitCredentials":{"password":"","username":""},"region":"","secretAccessKey":""}`,
 		},
+		{
+			name:           "second attempt of excluding same repo is ignored for Azure DevOps schema",
```
Comment (eseliger) [inline cmd/frontend/backend/external_services_test.go:50]: the name seems wrong?

File: cmd/frontend/backend/external_services_test.go [hunk 2]
```diff
@@ -58,6 +67,13 @@ func TestAddRepoToExclude(t *testing.T) {
 			initialConfig:  `{"repositoryQuery":["none"],"token":"abc","url":"https://bitbucket.sg.org","username":""}`,
 			expectedConfig: `{"exclude":[{"name":"SOURCEGRAPH/jsonrpc2"}],"repositoryQuery":["none"],"token":"abc","url":"https://bitbucket.sg.org","username":""}`,
 		},
+		{
+			name:           "second attempt of excluding same repo is ignored for Gerrit schema",
```
Comment (eseliger) [inline cmd/frontend/backend/external_services_test.go:71]: the name seems wrong?

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/backend/external_services_test.go (hunk 1 above)
Comment (pjlast) [reply cmd/frontend/backend/external_services_test.go:50]: Ugh copy pasting 🙄 thanks

Comment (pjlast) [reply cmd/frontend/backend/external_services_test.go:50]: I don't think so? 🤔  it's the name for all of these tests. They try to exclude the repo twice and confirm it's only excluded once

**Pull request merged**